numpy/ma/core.py

```diff
@@ -786,22 +786,17 @@ def fix_invalid(a, mask=nomask, copy=True, fill_value=None):
 
 class _DomainCheckInterval(object):
     """
-    Define a valid interval, so that :
-
-    ``domain_check_interval(a,b)(x) == True`` where
-    ``x < a`` or ``x > b``.
+    Define a valid interval, so that:
 
+    ``_DomainCheckInterval(a,b)(x)`` is False where ``a <= x <= b``.
```

Comment:
Hmm, you're right about `nan`s - this should read _"where ... or `isnan(x)`"_, right?

I think that's actually a bug in the existing code - that means all functions consider `nan` in their domain, which seems wrong to me. I don't know whether this is actually externally visible